### ISSUE TITLE
Error during install pytest-pycodestyle from PyPI

### ISSUE THREAD

**Issue by kr41** [henry0312/pytest-codestyle#21]
I see the project is available under two different names on PyPI `pytest-codestyle` and `pytest-pycodestyle`. However, the last one is not correctly installed. Here steps to reproduce:

1. Create new virtualenv
```
krat@krat-laptop:~/Projects$ virtualenv -p /usr/bin/python3.6 sandbox
Running virtualenv with interpreter /usr/bin/python3.6
Using base prefix '/usr'
New python executable in /home/krat/Projects/sandbox/bin/python3.6
Also creating executable in /home/krat/Projects/sandbox/bin/python
Installing setuptools, pkg_resources, pip, wheel...done.
krat@krat-laptop:~/Projects$ source sandbox/bin/activate
(sandbox) krat@krat-laptop:~/Projects$ pip --version
pip 9.0.3 from /home/krat/Projects/sandbox/lib/python3.6/site-packages (python 3.6)
(sandbox) krat@krat-laptop:~/Projects$ python --version
Python 3.6.2
```

2. `pytest-pycodestyle` installation crashes.
```
(sandbox) krat@krat-laptop:~/Projects$ pip install pytest-pycodestyle
Collecting pytest-pycodestyle
Collecting pycodestyle<2.4,>=2.3 (from pytest-pycodestyle)
  Using cached pycodestyle-2.3.1-py2.py3-none-any.whl
Collecting py<1.6,>=1.5 (from pytest-pycodestyle)
  Using cached py-1.5.3-py2.py3-none-any.whl
Collecting pytest<4,>=3.0 (from pytest-pycodestyle)
  Using cached pytest-3.5.0-py2.py3-none-any.whl
Collecting attrs>=17.4.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Using cached attrs-17.4.0-py2.py3-none-any.whl
Collecting more-itertools>=4.0.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Using cached more_itertools-4.1.0-py3-none-any.whl
Collecting six>=1.10.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Using cached six-1.11.0-py2.py3-none-any.whl
Collecting pluggy<0.7,>=0.5 (from pytest<4,>=3.0->pytest-pycodestyle)
Requirement already satisfied: setuptools in ./sandbox/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-pycodestyle)
Installing collected packages: pycodestyle, py, attrs, six, more-itertools, pluggy, pytest, pytest-pycodestyle
Exception:
Traceback (most recent call last):
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/wheel.py", line 347, in move_wheel_files
    assert info_dir, "%s .dist-info directory not found" % req
AssertionError: pytest-pycodestyle .dist-info directory not found
```

3. `pytest-codestyle` installation works OK.
```
(sandbox) krat@krat-laptop:~/Projects$ pip install pytest-codestyle
Requirement already satisfied: pytest-codestyle in ./sandbox/lib/python3.6/site-packages
Requirement already satisfied: py<1.6,>=1.5 in ./sandbox/lib/python3.6/site-packages (from pytest-codestyle)
Requirement already satisfied: pycodestyle<2.4,>=2.3 in ./sandbox/lib/python3.6/site-packages (from pytest-codestyle)
Requirement already satisfied: pytest<4,>=3.0 in ./sandbox/lib/python3.6/site-packages (from pytest-codestyle)
Requirement already satisfied: attrs>=17.4.0 in ./sandbox/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-codestyle)
Requirement already satisfied: setuptools in ./sandbox/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-codestyle)
Requirement already satisfied: six>=1.10.0 in ./sandbox/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-codestyle)
Requirement already satisfied: pluggy<0.7,>=0.5 in ./sandbox/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-codestyle)
Requirement already satisfied: more-itertools>=4.0.0 in ./sandbox/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-codestyle)
(sandbox) krat@krat-laptop:~/Projects$ 
```

**Comment by dmtucker** [comment]
I hit this last night (locally) also and thought it was a fluke because [Travis didn't repro](https://travis-ci.org/dmtucker/backlog/jobs/358755806#L485). Did something on PyPI change?

**Comment by henry0312** [comment]
Please try the below and tell me the result.

```
pip install --no-cache --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pytest-pycodestyle
```

**Comment by Enteris** [comment]
This line works
`pip install --no-cache --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pytest-pycodestyle`

**Comment by henry0312** [comment]
@Enteris Can you try it with virtualenv?

**Comment by henry0312** [comment]
Can anyone try it with virtualenv or pipenv?

**Comment by dmtucker** [comment]
WFM
```
david@kahuna:/tmp $ pipenv shell
Creating a virtualenv for this project…
Using /usr/bin/python3 (3.6.5) to create virtualenv…
⠋Already using interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/bin/python3
Also creating executable in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/bin/python
Installing setuptools, pip, wheel...done.

Virtualenv location: /home/david/.local/share/virtualenvs/tmp-XVr6zr33
Creating a Pipfile for this project…
Spawning environment shell (/bin/bash). Use 'exit' to leave.
. /home/david/.local/share/virtualenvs/tmp-XVr6zr33/bin/activate
david@kahuna:/tmp $ . /home/david/.local/share/virtualenvs/tmp-XVr6zr33/bin/activate
(tmp-XVr6zr33) david@kahuna:/tmp $ pip install --no-cache --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pytest-pycodestyle
Collecting pytest-pycodestyle
  Downloading https://test-files.pythonhosted.org/packages/b1/5d/f94c01e621d0078055c89cf5b0acc227c29403714bf574f05fd0a172ab30/pytest-pycodestyle-1.2.0.tar.gz
Collecting pytest<4,>=3.0 (from pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/ed/96/271c93f75212c06e2a7ec3e2fa8a9c90acee0a4838dc05bf379ea09aae31/pytest-3.5.0-py2.py3-none-any.whl (194kB)
    100% |████████████████████████████████| 194kB 2.8MB/s 
Collecting py<1.6,>=1.5 (from pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/67/a5/f77982214dd4c8fd104b066f249adea2c49e25e8703d284382eb5e9ab35a/py-1.5.3-py2.py3-none-any.whl (84kB)
    100% |████████████████████████████████| 92kB 3.5MB/s 
Collecting pycodestyle<2.4,>=2.3 (from pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/e4/81/78fe51eb4038d1388b7217dd63770b0f428370207125047312886c923b26/pycodestyle-2.3.1-py2.py3-none-any.whl (45kB)
    100% |████████████████████████████████| 51kB 4.5MB/s 
Requirement already satisfied: setuptools in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-pycodestyle)
Collecting attrs>=17.4.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/b5/60/4e178c1e790fd60f1229a9b3cb2f8bc2f4cc6ff2c8838054c142c70b5adc/attrs-17.4.0-py2.py3-none-any.whl
Collecting more-itertools>=4.0.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/7a/46/886917c6a4ce49dd3fff250c01c5abac5390d57992751384fe61befc4877/more_itertools-4.1.0-py3-none-any.whl (47kB)
    100% |████████████████████████████████| 51kB 5.0MB/s 
Collecting pluggy<0.7,>=0.5 (from pytest<4,>=3.0->pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/11/bf/cbeb8cdfaffa9f2ea154a30ae31a9d04a1209312e2919138b4171a1f8199/pluggy-0.6.0.tar.gz
Collecting six>=1.10.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
Installing collected packages: attrs, py, six, more-itertools, pluggy, pytest, pycodestyle, pytest-pycodestyle
  Running setup.py install for pluggy ... done
  Running setup.py install for pytest-pycodestyle ... done
Successfully installed attrs-17.4.0 more-itertools-4.1.0 pluggy-0.6.0 py-1.5.3 pycodestyle-2.3.1 pytest-3.5.0 pytest-pycodestyle-1.2.0 six-1.11.0
```

**Comment by Enteris** [comment]
I did it in virtualenv. @henry0312 
Here is the log : 

enteris@ubuntu:~/IdeaProjects/uam$ mkvirtualenv --python=/usr/bin/python3.6 pytest-codestyle-test
Running virtualenv with interpreter /usr/bin/python3.6
Using base prefix '/usr'
New python executable in /home/enteris/.virtualenvs/pytest-codestyle-test/bin/python3.6
Also creating executable in /home/enteris/.virtualenvs/pytest-codestyle-test/bin/python
Installing setuptools, pkg_resources, pip, wheel...done.
(pytest-codestyle-test) enteris@ubuntu:~/IdeaProjects/uam$ pip install --no-cache --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pytest-pycodestyle
Collecting pytest-pycodestyle
  Downloading https://test-files.pythonhosted.org/packages/b1/5d/f94c01e621d0078055c89cf5b0acc227c29403714bf574f05fd0a172ab30/pytest-pycodestyle-1.2.0.tar.gz
Collecting pytest<4,>=3.0 (from pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/ed/96/271c93f75212c06e2a7ec3e2fa8a9c90acee0a4838dc05bf379ea09aae31/pytest-3.5.0-py2.py3-none-any.whl (194kB)
    100% |████████████████████████████████| 194kB 4.7MB/s 
Collecting py<1.6,>=1.5 (from pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/67/a5/f77982214dd4c8fd104b066f249adea2c49e25e8703d284382eb5e9ab35a/py-1.5.3-py2.py3-none-any.whl (84kB)
    100% |████████████████████████████████| 92kB 6.4MB/s 
Collecting pycodestyle<2.4,>=2.3 (from pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/e4/81/78fe51eb4038d1388b7217dd63770b0f428370207125047312886c923b26/pycodestyle-2.3.1-py2.py3-none-any.whl (45kB)
    100% |████████████████████████████████| 51kB 11.5MB/s 
Collecting pluggy<0.7,>=0.5 (from pytest<4,>=3.0->pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/11/bf/cbeb8cdfaffa9f2ea154a30ae31a9d04a1209312e2919138b4171a1f8199/pluggy-0.6.0.tar.gz
Collecting more-itertools>=4.0.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/7a/46/886917c6a4ce49dd3fff250c01c5abac5390d57992751384fe61befc4877/more_itertools-4.1.0-py3-none-any.whl (47kB)
    100% |████████████████████████████████| 51kB 9.9MB/s 
Collecting six>=1.10.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools in /home/enteris/.virtualenvs/pytest-codestyle-test/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-pycodestyle)
Collecting attrs>=17.4.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Downloading https://files.pythonhosted.org/packages/b5/60/4e178c1e790fd60f1229a9b3cb2f8bc2f4cc6ff2c8838054c142c70b5adc/attrs-17.4.0-py2.py3-none-any.whl
Installing collected packages: pluggy, six, more-itertools, py, attrs, pytest, pycodestyle, pytest-pycodestyle
  Running setup.py install for pluggy ... done
  Running setup.py install for pytest-pycodestyle ... done
Successfully installed attrs-17.4.0 more-itertools-4.1.0 pluggy-0.6.0 py-1.5.3 pycodestyle-2.3.1 pytest-3.5.0 pytest-pycodestyle-1.2.0 six-1.11.0

**Comment by kr41** [comment]
Even `--no-cache` is enough:
```
krat@krat-laptop:~/Projects$ virtualenv -p /usr/bin/python3.6 sandbox
Running virtualenv with interpreter /usr/bin/python3.6
Using base prefix '/usr'
New python executable in /home/krat/Projects/sandbox/bin/python3.6
Also creating executable in /home/krat/Projects/sandbox/bin/python
Installing setuptools, pkg_resources, pip, wheel...done.
krat@krat-laptop:~/Projects$ source sandbox/bin/activate
(sandbox) krat@krat-laptop:~/Projects$ pip install --no-cache  pytest-pycodestyle
Collecting pytest-pycodestyle
  Downloading pytest-pycodestyle-1.2.0.tar.gz
  Running setup.py (path:/tmp/pip-build-nsi3joj4/pytest-pycodestyle/setup.py) egg_info for package pytest-pycodestyle produced metadata for project name pytest-codestyle. Fix your #egg=pytest-pycodestyle fragments.
Collecting pytest<4,>=3.0 (from pytest-codestyle)
  Downloading pytest-3.5.0-py2.py3-none-any.whl (194kB)
    100% |████████████████████████████████| 194kB 2.2MB/s 
Collecting py<1.6,>=1.5 (from pytest-codestyle)
  Downloading py-1.5.3-py2.py3-none-any.whl (84kB)
    100% |████████████████████████████████| 92kB 3.7MB/s 
Collecting pycodestyle<2.4,>=2.3 (from pytest-codestyle)
  Downloading pycodestyle-2.3.1-py2.py3-none-any.whl (45kB)
    100% |████████████████████████████████| 51kB 3.4MB/s 
Collecting six>=1.10.0 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading six-1.11.0-py2.py3-none-any.whl
Collecting pluggy<0.7,>=0.5 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading pluggy-0.6.0.tar.gz
Collecting attrs>=17.4.0 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading attrs-17.4.0-py2.py3-none-any.whl
Collecting more-itertools>=4.0.0 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading more_itertools-4.1.0-py3-none-any.whl (47kB)
    100% |████████████████████████████████| 51kB 3.9MB/s 
Requirement already satisfied: setuptools in ./sandbox/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-codestyle)
Installing collected packages: six, pluggy, attrs, py, more-itertools, pytest, pycodestyle, pytest-codestyle
  Running setup.py install for pluggy ... done
  Running setup.py install for pytest-codestyle ... done
Successfully installed attrs-17.4.0 more-itertools-4.1.0 pluggy-0.6.0 py-1.5.3 pycodestyle-2.3.1 pytest-3.5.0 pytest-codestyle-1.2.0 six-1.11.0
```

But without any options it doesn't work:
```
(sandbox) krat@krat-laptop:~/Projects$ deactivate 
krat@krat-laptop:~/Projects$ rm -r sandbox/
krat@krat-laptop:~/Projects$ virtualenv -p /usr/bin/python3.6 sandbox
Running virtualenv with interpreter /usr/bin/python3.6
Using base prefix '/usr'
New python executable in /home/krat/Projects/sandbox/bin/python3.6
Also creating executable in /home/krat/Projects/sandbox/bin/python
Installing setuptools, pkg_resources, pip, wheel...done.
krat@krat-laptop:~/Projects$ source sandbox/bin/activate
(sandbox) krat@krat-laptop:~/Projects$ pip install pytest-pycodestyle
Collecting pytest-pycodestyle
Collecting pytest<4,>=3.0 (from pytest-pycodestyle)
  Using cached pytest-3.5.0-py2.py3-none-any.whl
Collecting pycodestyle<2.4,>=2.3 (from pytest-pycodestyle)
  Using cached pycodestyle-2.3.1-py2.py3-none-any.whl
Collecting py<1.6,>=1.5 (from pytest-pycodestyle)
  Using cached py-1.5.3-py2.py3-none-any.whl
Collecting pluggy<0.7,>=0.5 (from pytest<4,>=3.0->pytest-pycodestyle)
Requirement already satisfied: setuptools in ./sandbox/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-pycodestyle)
Collecting attrs>=17.4.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Using cached attrs-17.4.0-py2.py3-none-any.whl
Collecting six>=1.10.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Using cached six-1.11.0-py2.py3-none-any.whl
Collecting more-itertools>=4.0.0 (from pytest<4,>=3.0->pytest-pycodestyle)
  Using cached more_itertools-4.1.0-py3-none-any.whl
Installing collected packages: pluggy, attrs, py, six, more-itertools, pytest, pycodestyle, pytest-pycodestyle
Exception:
Traceback (most recent call last):
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/home/krat/Projects/sandbox/lib/python3.6/site-packages/pip/wheel.py", line 347, in move_wheel_files
    assert info_dir, "%s .dist-info directory not found" % req
AssertionError: pytest-pycodestyle .dist-info directory not found
(sandbox) krat@krat-laptop:~/Projects$
```

**Comment by dmtucker** [comment]
That's interesting about `--no-cache-dir`! If I delete my cache and try again, I get a different error:
```
david@kahuna:/tmp $ rm -rf ~/.cache/pip*
david@kahuna:/tmp $ pipenv run pip install pytest-pycodestyle
Collecting pytest-pycodestyle
  Downloading pytest-pycodestyle-1.2.0.tar.gz
  Running setup.py (path:/tmp/pip-build-qfxz1_s7/pytest-pycodestyle/setup.py) egg_info for package pytest-pycodestyle produced metadata for project name pytest-codestyle. Fix your #egg=pytest-pycodestyle fragments.
Collecting pytest<4,>=3.0 (from pytest-codestyle)
  Downloading pytest-3.5.0-py2.py3-none-any.whl (194kB)
    100% |████████████████████████████████| 194kB 1.5MB/s 
Collecting py<1.6,>=1.5 (from pytest-codestyle)
  Downloading py-1.5.3-py2.py3-none-any.whl (84kB)
    100% |████████████████████████████████| 92kB 2.7MB/s 
Collecting pycodestyle<2.4,>=2.3 (from pytest-codestyle)
  Downloading pycodestyle-2.3.1-py2.py3-none-any.whl (45kB)
    100% |████████████████████████████████| 51kB 4.3MB/s 
Collecting pluggy<0.7,>=0.5 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading pluggy-0.6.0.tar.gz
Collecting attrs>=17.4.0 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading attrs-17.4.0-py2.py3-none-any.whl
Collecting six>=1.10.0 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading six-1.11.0-py2.py3-none-any.whl
Collecting more-itertools>=4.0.0 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading more_itertools-4.1.0-py3-none-any.whl (47kB)
    100% |████████████████████████████████| 51kB 3.1MB/s 
Requirement already satisfied: setuptools in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-codestyle)
Building wheels for collected packages: pytest-codestyle, pytest-codestyle, pluggy
  Running setup.py bdist_wheel for pytest-codestyle ... done
  Stored in directory: /home/david/.cache/pip/wheels/5d/13/7b/f794d2c06ce49a189d8107be57223f387f87c94c553fe7e3cd
  Running setup.py bdist_wheel for pytest-codestyle ... error
  Complete output from command /home/david/.local/share/virtualenvs/tmp-XVr6zr33/bin/python3.6m -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-qfxz1_s7/pytest-codestyle/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpqq9duk6vpip-wheel- --python-tag cp36:
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/tokenize.py", line 452, in open
      buffer = _builtin_open(filename, 'rb')
  FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pip-build-qfxz1_s7/pytest-codestyle/setup.py'
  
  ----------------------------------------
  Failed building wheel for pytest-codestyle
  Running setup.py clean for pytest-codestyle
  Complete output from command /home/david/.local/share/virtualenvs/tmp-XVr6zr33/bin/python3.6m -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-qfxz1_s7/pytest-codestyle/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" clean --all:
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/tokenize.py", line 452, in open
      buffer = _builtin_open(filename, 'rb')
  FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pip-build-qfxz1_s7/pytest-codestyle/setup.py'
  
  ----------------------------------------
  Failed cleaning build dir for pytest-codestyle
  Running setup.py bdist_wheel for pluggy ... done
  Stored in directory: /home/david/.cache/pip/wheels/df/44/8e/e136760ae525eac46b3e3db643ef58ff1753177b5a722b0c96
Successfully built pytest-codestyle pluggy
Failed to build pytest-codestyle
Installing collected packages: pluggy, attrs, six, more-itertools, py, pytest, pycodestyle, pytest-codestyle
Successfully installed attrs-17.4.0 more-itertools-4.1.0 pluggy-0.6.0 py-1.5.3 pycodestyle-2.3.1 pytest-3.5.0 pytest-codestyle-1.2.0 six-1.11.0
david@kahuna:/tmp $ pipenv run pip install pytest-pycodestyle
Collecting pytest-pycodestyle
Requirement already satisfied: pytest<4,>=3.0 in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages (from pytest-pycodestyle)
Requirement already satisfied: pycodestyle<2.4,>=2.3 in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages (from pytest-pycodestyle)
Requirement already satisfied: py<1.6,>=1.5 in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages (from pytest-pycodestyle)
Requirement already satisfied: attrs>=17.4.0 in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-pycodestyle)
Requirement already satisfied: more-itertools>=4.0.0 in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-pycodestyle)
Requirement already satisfied: pluggy<0.7,>=0.5 in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-pycodestyle)
Requirement already satisfied: six>=1.10.0 in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-pycodestyle)
Requirement already satisfied: setuptools in /home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-pycodestyle)
Installing collected packages: pytest-pycodestyle
Exception:
Traceback (most recent call last):
  File "/home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/home/david/.local/share/virtualenvs/tmp-XVr6zr33/lib/python3.6/site-packages/pip/wheel.py", line 347, in move_wheel_files
    assert info_dir, "%s .dist-info directory not found" % req
AssertionError: pytest-pycodestyle .dist-info directory not found
[Thu Mar 29 21:55:03 PDT 2018] 2
```

**Comment by dmtucker** [comment]
Maybe that's a pipenv issue?
```
david@kahuna:/tmp $ pipenv --rm
Removing virtualenv (/home/david/.local/share/virtualenvs/tmp-XVr6zr33)…
david@kahuna:/tmp $ rm -rf ~/.cache/pip*
david@kahuna:/tmp $ python3 -m venv tmpvenv
david@kahuna:/tmp $ tmpvenv/bin/pip install --upgrade pip setuptools
Collecting pip
  Downloading pip-9.0.3-py2.py3-none-any.whl (1.4MB)
    100% |████████████████████████████████| 1.4MB 1.2MB/s 
Collecting setuptools
  Downloading setuptools-39.0.1-py2.py3-none-any.whl (569kB)
    100% |████████████████████████████████| 573kB 2.2MB/s 
Installing collected packages: pip, setuptools
  Found existing installation: pip 9.0.1
    Uninstalling pip-9.0.1:
      Successfully uninstalled pip-9.0.1
  Found existing installation: setuptools 32.3.1
    Uninstalling setuptools-32.3.1:
      Successfully uninstalled setuptools-32.3.1
Successfully installed pip-9.0.3 setuptools-39.0.1
david@kahuna:/tmp $ tmpvenv/bin/pip install pytest-pycodestyle
Collecting pytest-pycodestyle
  Downloading pytest-pycodestyle-1.2.0.tar.gz
  Running setup.py (path:/tmp/pip-build-yqm90ktk/pytest-pycodestyle/setup.py) egg_info for package pytest-pycodestyle produced metadata for project name pytest-codestyle. Fix your #egg=pytest-pycodestyle fragments.
Collecting pytest<4,>=3.0 (from pytest-codestyle)
  Downloading pytest-3.5.0-py2.py3-none-any.whl (194kB)
    100% |████████████████████████████████| 194kB 2.2MB/s 
Collecting py<1.6,>=1.5 (from pytest-codestyle)
  Downloading py-1.5.3-py2.py3-none-any.whl (84kB)
    100% |████████████████████████████████| 92kB 2.8MB/s 
Collecting pycodestyle<2.4,>=2.3 (from pytest-codestyle)
  Downloading pycodestyle-2.3.1-py2.py3-none-any.whl (45kB)
    100% |████████████████████████████████| 51kB 5.3MB/s 
Collecting more-itertools>=4.0.0 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading more_itertools-4.1.0-py3-none-any.whl (47kB)
    100% |████████████████████████████████| 51kB 4.0MB/s 
Collecting attrs>=17.4.0 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading attrs-17.4.0-py2.py3-none-any.whl
Collecting pluggy<0.7,>=0.5 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading pluggy-0.6.0.tar.gz
Collecting six>=1.10.0 (from pytest<4,>=3.0->pytest-codestyle)
  Downloading six-1.11.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools in ./tmpvenv/lib/python3.6/site-packages (from pytest<4,>=3.0->pytest-codestyle)
Installing collected packages: py, six, more-itertools, attrs, pluggy, pytest, pycodestyle, pytest-codestyle
  Running setup.py install for pluggy ... done
  Running setup.py install for pytest-codestyle ... done
Successfully installed attrs-17.4.0 more-itertools-4.1.0 pluggy-0.6.0 py-1.5.3 pycodestyle-2.3.1 pytest-3.5.0 pytest-codestyle-1.2.0 six-1.11.0
```
@kr41 Have you tried clearing `pip`'s cache (e.g. `rm -rf ~/.cache/pip*`)?

**Comment by henry0312** [comment]
I released a new test version. Can anyone try to install it?

```sh
pip install --no-cache --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pytest-pycodestyle==1.2.1
```

and 

```sh
pip install --no-cache --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pytest-codestyle==1.2.1
```

**Comment by henry0312** [comment]
I uploaded 1.2.1.dev1, please try it.

**Comment by dmtucker** [comment]
WFM :+1:

**Comment by henry0312** [comment]
Thanks.
I'll release a fixed version.

**Comment by henry0312** [comment]
I just released `v1.2.1` on PyPI (not TestPyPI).
It should be fixed, please try it.

**Comment by kr41** [comment]
@henry0312 It works, thank you!
